packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift

```diff
@@ -0,0 +1,425 @@
+    _ applicationUserName: String?, error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    paymentQueueHandler?.restoreTransactions(applicationUserName)
+  }
+
+  public func presentCodeRedemptionSheetWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    #if os(iOS)
+      paymentQueueHandler!.presentCodeRedemptionSheet()
+    #endif
+  }
+
+  public func retrieveReceiptDataWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) -> String? {
+    var flutterError: FlutterError? = nil
+    let receiptData: String? = receiptManager.retrieveReceiptWithError(&flutterError)
+    if receiptData == nil {
+      error.pointee = flutterError
+      return nil
+    }
+    return receiptData
+  }
+
+  public func refreshReceiptReceiptProperties(
+    _ receiptProperties: [String: Any]?, completion: @escaping (FlutterError?) -> Void
+  ) {
+    var request: SKReceiptRefreshRequest
+    if let receiptProperties = receiptProperties {
+      // If receiptProperties is not nil, this call is for testing.
+      let properties = [
+        SKReceiptPropertyIsExpired: receiptProperties["isExpired"]!,
+        SKReceiptPropertyIsRevoked: receiptProperties["isRevoked"]!,
+        SKReceiptPropertyIsVolumePurchase: receiptProperties["isVolumePurchase"]!,
+      ]
+      request = getRefreshReceiptRequest(properties: properties)
+    } else {
+      request = getRefreshReceiptRequest(properties: nil)
+    }
+
+    let handler = handlerFactory(request)
+    requestHandlers.insert(handler)
+    handler.startProductRequest { [weak self] response, error in
+      if let error = error {
+        let requestError = FlutterError(
+          code: "storekit_refreshreceiptrequest_platform_error",
+          message: error.localizedDescription,
+          details: error.localizedDescription)
+        completion(requestError)
+        return
+      }
+      completion(nil)
+      self?.requestHandlers.remove(handler)
+    }
+  }
+
+  public func startObservingPaymentQueueWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    paymentQueueHandler!.startObservingPaymentQueue()
+  }
+
+  public func stopObservingPaymentQueueWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    paymentQueueHandler!.stopObservingPaymentQueue()
+  }
+
+  public func registerPaymentQueueDelegateWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    #if os(iOS)
+      if #available(iOS 13.0, *) {
+        let messenger = registrar?.messenger()
+        paymentQueueDelegateCallbackChannel = FlutterMethodChannel(
+          name: "plugins.flutter.io/in_app_purchase_payment_queue_delegate",
+          binaryMessenger: messenger!)
+
+        paymentQueueDelegate = FIAPPaymentQueueDelegate(
+          methodChannel: paymentQueueDelegateCallbackChannel!)
+        paymentQueueHandler!.delegate = (paymentQueueDelegate as! any SKPaymentQueueDelegate)
```

Comment:
iirc `any` keyword is quite recent. probably a few years after ios 13. can you double check?

Reply:
turns out i can just delete it lol